### ISSUE TITLE
[DAY 24] 정점 표현 & 추천시스템 (심화)

### ISSUE THREAD

**Issue by changwoomon** [boostcamp-ai-tech-4/peer-session#89]
### ❓질문

`MJ`

* 임의보행 기반 접근법에서 `임의보행 중 마주친 모든 정점에 대한 합산`이 언제까지 돌리는 건가요?
* 네거티브 샘플링은 V만 건드리는 것이 맞나요?
  * [1](https://ddiri01.tistory.com/310), [2](https://wikidocs.net/69141) -> 자연어 처리 쪽의 네거티브 샘플링 같은데 관련이 있을까요?
* 네거티브 샘플링에서 임의의 u만 뽑아온다면 나머지에 대한 임베딩벡터는 어떻게 구하나요?

  <img src="https://user-images.githubusercontent.com/59998179/109281539-c955c600-785f-11eb-9a94-54016c7c5ada.png" width="550" height="250">

`펭귄`

* 8강 p.24에서 `과적합이 발생할 수 있다`는 의미가 무엇인가요?
  * 극단적으로 학습되는 것을 방지하기 위함입니다.
  * 노이즈에 대해서도 오버피팅을 하기 때문에 정규화도 같이 해주는 것입니다.
* 모형 복잡도 부분 : 릿지, 라쏘 회귀 / weight decay
  * 패널티를 줘서 완전히 피팅되지 않고 적절히 피팅되도록 하는 것입니다.
  * p, q가 0으로 돼야 loss function이 줄어드니까 원점으로 당겨지는 것이 아닐까요?
  * [Ridge regression(릿지 회귀)와 Lasso regression(라쏘 회귀) 쉽게 이해하기](https://rk1993.tistory.com/entry/Ridge-regression%EC%99%80-Lasso-regression-%EC%89%BD%EA%B2%8C-%EC%9D%B4%ED%95%B4%ED%95%98%EA%B8%B0)
  * [선형 회귀(linear regression) 그리고 라쏘(Lasso)](https://bskyvision.com/193)

  <img src="https://user-images.githubusercontent.com/59998179/109285372-5864dd00-7864-11eb-8f9b-cabb75611805.png" width="550" height="250">

---

### 🤷‍♀️Further Question🤷‍♀️

* 추천시스템의 성능을 측정하는 metric이 RMSE라는 것은 예상 평점이 높은 상품과 낮은 상품에 동일한 페널티를 부여한다는 것을 뜻합니다. 하지만 실제로 추천시스템에서는 내가 좋아할 것 같은 상품을 추천해주는것, 즉 예상 평점이 높은 상품을 잘 맞추는것이 중요합니다. 이를 고려하여 성능을 측정하기 위해서는 어떻게 해야 할까요?
  * 전체 한 번 학습한 후 높은 평점에 관한 데이터만 뽑아서 다시 학습한다.
  * 여러 모델로 학습해서 나온 결과를 ensemble 할 때 가중치를 준다. (높은 점수에 힘을 실어주자)
  * P, Q를 곱할 때 큰 가중치가 더해지도록 r에 비례하도록 한다.
  * 질문이 정확히 뭘까요..?
    * 동일한 정규화?
    * 높은 평점일 걸 낮게 평가하는거?

* 추천 시스템의 성능을 향상시키기 위해서는 어떠한 것을 더 고려할 수 있을까요? (해당 문제는 정답이 제공되지 않는 문제입니다. 자유롭게 여러분의 의견을 이야기해보세요.)
  * 평가되지 않은 영화들에 대해서는 메타 데이터를 이용하여 선호도를 대충 추측한다. 사용 전 선호도는 아이템의 메타 데이터(감독, 배우 등)을, 사용 후 선호도는 아이템 자체의 내용을 기반으로 한다.
  * 사용 전 선호도와 uninteresting item은 어떤 관계가 있을까?
    * 사용 전 선호도는 평점을 매긴 item을 기반으로 한다. unrated item을 사용 전 선호도를 기반으로 파악할 수 있다. 이것을 이용해 임계값을 넘지 못하면 다 0점 처리하는 Zero Injection 방법을 사용한다.

**Comment by changwoomon** [comment]
Ans. Recall@k, Precisoin@k, MAP, MRR, NDCG 등의 랭킹 기반 메트릭을 활용한다.
